src/cluster/k8sClientHandler.go

```diff
@@ -36,14 +39,62 @@ func isInCluster() bool {
 }
 
 func ConnectK8sClient() *kubernetes.Clientset {
-	if isInCluster() {
-		return ConnectInClusterAPIClient()
+	config, err := GetConfig()
+	if err != nil {
+		log.Error().Msg(err.Error())
+		return nil
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Error().Msg(err.Error())
+		return nil
 	}
 
-	return ConnectLocalAPIClient()
+	return clientset
 }
 
-func ConnectLocalAPIClient() *kubernetes.Clientset {
+// GetConfig func
+func GetConfig() (*rest.Config, error) {
+	if kconfig != nil {
+		return kconfig, nil
+	}
+	if isInCluster() {
+		host := ""
+		port := ""
+		token := ""
+
+		if val, ok := os.LookupEnv("KUBERNETES_SERVICE_HOST"); ok {
+			host = val
+		} else {
+			host = "127.0.0.1"
+		}
+
+		if val, ok := os.LookupEnv("KUBERNETES_PORT_443_TCP_PORT"); ok {
+			port = val
+		} else {
+			port = "6443"
+		}
+
+		read, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+		if err != nil {
+			log.Error().Msg(err.Error())
+			return nil, err
+		}
+
+		token = string(read)
+
+		// create the configuration by token
+		kconfig = &rest.Config{
+			Host:        "https://" + host + ":" + port,
+			BearerToken: token,
+			TLSClientConfig: rest.TLSClientConfig{
+				Insecure: true,
+			},
+		}
+
+		return kconfig, nil
+	}
+
```

Comment:
can you please use `sigs.k8s.io/controller-runtime` GetConfigOrDie function, it is more convenient.

Reply:
Used `sigs.k8s.io/controller-runtime` GetConfig function here. just wanted to handle error instead of exit in a particular case.